compiler/rustc_middle/src/ty/flags.rs

```diff
@@ -59,6 +59,10 @@ impl FlagComputation {
     {
         let mut computation = FlagComputation::new();
 
+        if !value.bound_vars().is_empty() {
```

Comment:
Let's have a comment about this. Something like:

In some cases, there are binders with variables that are unused (e.g., `for<'a> fn(u32)`). Set the flag to represent the `'a` in this example.